lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -453,21 +476,66 @@ static SILValue reapplyFunctionConversion(
   llvm_unreachable("Unhandled function conversion instruction");
 }
 
-/// Emits a reference to a derivative function of `original`, differentiated
-/// with respect to a superset of `desiredIndices`. Returns the `SILValue` for
-/// the derivative function and the actual indices that the derivative function
-/// is with respect to.
-///
-/// Returns `None` on failure, signifying that a diagnostic has been emitted
-/// using `invoker`.
-static std::optional<std::pair<SILValue, AutoDiffConfig>>
-emitDerivativeFunctionReference(
-    DifferentiationTransformer &transformer, SILBuilder &builder,
-    const AutoDiffConfig &desiredConfig, AutoDiffDerivativeFunctionKind kind,
-    SILValue original, DifferentiationInvoker invoker,
-    SmallVectorImpl<AllocStackInst *> &newBuffersToDealloc) {
-  ADContext &context = transformer.getContext();
+SILFunction *DifferentiationTransformer::getUnwrappedCurryThunkFunction(
+    SILFunction *originalFn) {
+  auto *autoCE = dyn_cast_or_null<AutoClosureExpr>(
+      originalFn->getDeclRef().getAbstractClosureExpr());
+  if (autoCE == nullptr)
+    return nullptr;
+
+  AbstractFunctionDecl *afd = nullptr;
+  Expr *expr = autoCE->getUnwrappedCurryThunkExpr();
+  switch (autoCE->getThunkKind()) {
+  default:
+    llvm_unreachable("Only single and double curry thunks are expected");
+  case AutoClosureExpr::Kind::SingleCurryThunk:
+  case AutoClosureExpr::Kind::DoubleCurryThunk:
+    afd = cast<AbstractFunctionDecl>(cast<ApplyExpr>(expr)->getCalledValue(
+        /*skipFunctionConversions=*/true));
+    break;
+  }
+  assert(afd);
+
+  auto silFnIt = afdToSILFn.find(afd);
+  if (silFnIt == afdToSILFn.end()) {
+    assert(afdToSILFn.empty());
```

Comment:
see above

Reply:
Addressed in 4d74d349573bed30a0f64fbc90de89b1392e1830, thanks